tooling/nargo_fmt/src/formatter/chunks.rs

```diff
@@ -0,0 +1,776 @@
+
+        if let Some(Chunk::Group(group)) = self.chunks.last_mut() {
+            group.text(chunk);
+        } else {
+            self.text(chunk);
+        }
+    }
+
+    /// Appends a trailing comment (it's formatted slightly differently than a regular text chunk).
+    pub(crate) fn trailing_comment(&mut self, chunk: TextChunk) {
+        if chunk.width > 0 {
+            self.push(Chunk::TrailingComment(chunk));
+        }
+    }
+
+    /// Appends a leading comment (it's formatted slightly differently than a regular text chunk).
+    pub(crate) fn leading_comment(&mut self, chunk: TextChunk) {
+        if chunk.width > 0 {
+            self.push(Chunk::LeadingComment(chunk));
+        }
+    }
+
+    /// Appends a trailing comma (will only show up if the group is formatted in multiple lines).
+    pub(crate) fn trailing_comma(&mut self) {
+        self.push(Chunk::TrailingComma);
+    }
+
+    /// Appends another group as a nested group.
+    pub(crate) fn group(&mut self, group: ChunkGroup) {
+        self.push(Chunk::Group(group));
+    }
+
+    /// Append one line to this chunk.
+    pub(crate) fn line(&mut self) {
+        self.lines(false);
+    }
+
+    /// Append one or two lines to this chunk.
+    pub(crate) fn lines(&mut self, two: bool) {
+        self.push(Chunk::Line { two });
+    }
+
+    /// Appends a SpaceOrLine chunk, which means that it's a space when this group is
+    /// formatted in a single line, or a line when it's formatted in multiple lines.
+    pub(crate) fn space_or_line(&mut self) {
+        self.push(Chunk::SpaceOrLine);
+    }
+
+    pub(crate) fn increase_indentation(&mut self) {
+        self.push(Chunk::IncreaseIndentation);
+    }
+
+    pub(crate) fn decrease_indentation(&mut self) {
+        self.push(Chunk::DecreaseIndentation);
+    }
+
+    pub(crate) fn push_indentation(&mut self) {
+        self.push(Chunk::PushIndentation);
+    }
+
+    pub(crate) fn pop_indentation(&mut self) {
+        self.push(Chunk::PopIndentation);
+    }
+
+    pub(crate) fn push(&mut self, chunk: Chunk) {
+        self.chunks.push(chunk);
+    }
+
+    pub(crate) fn width(&self) -> usize {
+        self.chunks.iter().map(|chunk| chunk.width()).sum()
+    }
+
+    pub(crate) fn expression_list_width(&self) -> usize {
+        if self.kind.is_method_call() {
+            self.chunks
+                .iter()
+                .map(|chunk| {
+                    if let Chunk::Group(group) = chunk {
+                        if group.kind == GroupKind::ExpressionList {
+                            group.expression_list_width()
+                        } else {
+                            chunk.width_inside_an_expression_list()
+                        }
+                    } else {
+                        chunk.width_inside_an_expression_list()
+                    }
+                })
+                .sum()
+        } else {
+            self.chunks.iter().map(|chunk| chunk.width_inside_an_expression_list()).sum()
```

Comment:
It's not clear to me why the special treatment of `GroupKind::ExpressionList` only applies to the `is_method_call()` case 🤔

Reply:
I just tried removing the `is_method_call` branch and all tests pass, no change is produced in the standard library, test programs or Aztec-Packages. Go apparently there's no need for this special case. I think in my head I thought there would be a difference 😅